File: services/api/app/schemas/couple_companion.py
"""Schemas: Paar-Begleiter mit Gesprächsverlauf und Zusammenfassungen.

Alles hier gehört ausschließlich der anfragenden Person — die Partnerperson hat keinen
Endpunkt, über den sie es sehen könnte.
"""
from __future__ import annotations

from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.couple_private import CouplePrivateMessage


class CoupleEchoThread(BaseModel):
    id: UUID
    title: str | None = None
    #: 'chat' = offenes Gespraech, 'deescalation' = nach einem Streit.
    kind: str = "chat"
    created_at: datetime
    updated_at: datetime
    closed_at: datetime | None = None
    message_count: int = 0
    summary_count: int = 0


class CoupleEchoConversation(BaseModel):
    """Ein Gespräch samt Verlauf."""
    thread: CoupleEchoThread
    messages: list[CouplePrivateMessage] = []


class CoupleEchoSummary(BaseModel):
    id: UUID
    thread_id: UUID | None = None
    title: str | None = None
    summary_text: str
    created_at: datetime
    updated_at: datetime


class CoupleEchoSummaryEdit(BaseModel):
    title: str | None = Field(None, max_length=160)
    summary_text: str | None = Field(None, min_length=1, max_length=6000)


class CoupleThreadRename(BaseModel):
    title: str = Field(..., min_length=1, max_length=160)


class CoupleSceneDraft(BaseModel):
    """Ein Szenen-ENTWURF aus einem eigenen Echo-Gespraech.

    Wird nicht gespeichert. Die nutzende Person prueft und bearbeitet ihn und legt ihn
    dann ueber den regulaeren Fall-Endpunkt an - der Paarbereich schreibt nie in einen Fall.
    """

    title: str = ""
    description: str = ""
    user_reaction: str | None = None
    scene_date: date | None = None
    #: 1-5 wie im Fall-Bereich.
    distress_score: int | None = None
    pattern_tags: list[str] = []

File: apps/web/src/api/coupleCompanion.ts
import { apiClient } from './client'
import type { CouplePrivateMessage } from './couplePrivate'

/**
 * Paar-Begleiter: Gespräche mit Verlauf und Zusammenfassungen.
 *
 * Dasselbe Vorgehen wie bei den Themendialogen im Fall-Bereich – reden, zusammenfassen
 * lassen, behalten. Alles hier gehört ausschließlich dir; die Partnerperson hat keinen
 * Endpunkt, über den sie es sehen könnte.
 */
/**
 * Art eines Fadens. Sie entscheidet ueber den Ton, in dem Echo antwortet, und haelt die
 * Faeden auseinander – ein Streit-Einstieg landet nie mitten in einem offenen Gespraech.
 */
export type CoupleThreadKind = 'chat' | 'deescalation'

export interface CoupleEchoThread {
  id: string
  title: string | null
  kind: CoupleThreadKind
  created_at: string
  updated_at: string
  closed_at: string | null
  message_count: number
  summary_count: number
}

export interface CoupleEchoConversation {
  thread: CoupleEchoThread
  messages: CouplePrivateMessage[]
}

export interface CoupleEchoSummary {
  id: string
  thread_id: string | null
  title: string | null
  summary_text: string
  created_at: string
  updated_at: string
}

export const coupleCompanionApi = {
  /** Das laufende Gespräch. */
  current: (coupleId: string, kind: CoupleThreadKind = 'chat') =>
    apiClient.get<CoupleEchoConversation>(`/couple/links/${coupleId}/echo`, { params: { kind } })
      .then(r => r.data),

  send: (coupleId: string, content: string, kind: CoupleThreadKind = 'chat') =>
    apiClient.post<CoupleEchoConversation>(`/couple/links/${coupleId}/echo`, { content },
      { params: { kind } }).then(r => r.data),

  /** Fasst zusammen, schließt das Gespräch ab und behält die Zusammenfassung. */
  summarize: (coupleId: string, kind: CoupleThreadKind = 'chat') =>
    apiClient.post<CoupleEchoSummary>(`/couple/links/${coupleId}/echo/summary`, null,
      { params: { kind } }).then(r => r.data),

  threads: (coupleId: string) =>
    apiClient.get<CoupleEchoThread[]>(`/couple/links/${coupleId}/echo/threads`).then(r => r.data),

  thread: (threadId: string) =>
    apiClient.get<CoupleEchoConversation>(`/couple/echo/threads/${threadId}`).then(r => r.data),

  summaries: (coupleId: string) =>
    apiClient.get<CoupleEchoSummary[]>(`/couple/links/${coupleId}/echo/summaries`).then(r => r.data),

  editSummary: (summaryId: string, body: { title?: string | null; summary_text?: string }) =>
    apiClient.patch<CoupleEchoSummary>(`/couple/echo/summaries/${summaryId}`, body).then(r => r.data),

  deleteSummary: (summaryId: string) =>
    apiClient.delete(`/couple/echo/summaries/${summaryId}`),
}
